Content.Server/Botany/Systems/PlantHolderSystem.cs

```diff
@@ -331,13 +376,29 @@ public void Update(EntityUid uid, PlantHolderComponent? component = null)
             }
 
             // Weeds like water and nutrients! They may appear even if there's not a seed planted.
-            if (component.WaterLevel > 10 && component.NutritionLevel > 2 && _random.Prob(component.Seed == null ? 0.05f : 0.01f))
+            if (component.WaterLevel > 10 && component.NutritionLevel > 2 &&
+                _random.Prob(component.Seed == null ? 0.05f : 0.01f))
             {
                 component.WeedLevel += 1 * HydroponicsSpeedMultiplier * component.WeedCoefficient;
 
                 if (component.DrawWarnings)
                     component.UpdateSpriteAfterUpdate = true;
             }
+            else if (component.WaterLevel > 10 && component.NutritionLevel > 2 && component.Seed != null &&
+                     component.Seed.TurnIntoKudzu)
+            {
+                component.WeedLevel += 8 * HydroponicsSpeedMultiplier * component.WeedCoefficient;
+
+                if (component.DrawWarnings)
+                    component.UpdateSpriteAfterUpdate = true;
+            }
+
+            if (component.Seed != null && component.Seed.TurnIntoKudzu && component.WeedLevel >= 10)
```

Comment:
store this in the component as `WeedHighLevelThreshold` or something and make this as well as the check in line 99 of this use that